docs/book/src/developer/providers/ipam.md

```diff
@@ -0,0 +1,81 @@
+
+IPAM providers are optional when using Cluster API. Infrastructure providers need to [implement explicit support](#infrastructure-provider) to be usable in conjunction with IPAM providers.
+
+<aside class="note">
+
+Note that the IPAM contract is single-stack. If you need both v4 and v6 addresses, two pools and two IPAddressClaims are necessary.
+
+</aside>
+
+## Data Types
+
+An IPAM provider must define one or more API types for IP address pools. The types:
+
+`1. Must belong to an API group served by the Kubernetes apiserver 
+2. Must be implemented as a CustomResourceDefinition. 
+   The CRD name must have the format produced by `sigs.k8s.io/cluster-api/util/contract.CalculateCRDName(Group, Kind)`. 
+3. Must have the standard Kubernetes "type metadata" and "object metadata"
+`4. Should have a status.conditions field with the following:
+   1. A Ready condition to represent the overall operational state of the component. It can be based on the summary of more detailed conditions existing on the same object, e.g. instanceReady, SecurityGroupsReady conditions.
+
+## Behaviour
+
+IPAM providers must handle any IPAddressClaim resources that reference IP address pools that are managed by the provider and create an IPAddress resource for it. IPAddressClaims are usually created by infrastructure providers.
+
+### IPAM Provider
+
+An IPAM provider must watch for new, updated and deleted IPAddressClaims that reference an IP address pool that is manged by the provider in their `spec.poolRef` field.
+
+#### Normal IPAddressClaim
+
+1. If the IPAddressClaim does not reference a pool managed by the provider in it's `spec.poolRef`, abort the reconciliation.
+2. If the related Cluster is paused, abort reconciliation
+   1. The related Cluster is referenced using the `spec.clusterName` field or a `cluster.x-k8s.io/cluster-name` label (the latter is deprecated).
+   2. If the field is empty and the annotation is not present, reconciliation can continue.
+   3. If the referenced cluster is not found, abort reconciliation.
+   4. If the referenced cluster has `spec.paused` set or a `cluster.x-k8s.io/paused` annotation, skip reconciliation
+3. Add any required provider-specific Finalziers (you probably need one)
+4. Allocate an IP address for the claim
+5. Create an IPAddress object
+   1. It should have the same name as the claim.
+   2. It must have a owner reference with `controller: true` and `blockOwnerDeletion: true` to the Claim
+   3. It must have a owner reference with `controller: false` and `blockOwnerDeletion: true` to the referenced Pool
+   4. It should have a Finalizer that prevents accidental deletion, e.g. `ipam.cluster.x-k8s.io/protect-address`.
+6. Set the `status.addressRef` to the created IPAddress
+
+#### Deleted IPAddressClaim
+
+1. If the related Cluster is paused, abort reconciliation (see 2. above)
+2. Deallocate the IP address
+3. Delete the IPAddress object
+   1. Remove any Finalizers that were set to prevent deletion first
```

Comment:
What do you mean with "prevent deletion first", more specifically what is the "first" referring to?
(I would probably just drop first)